### PR TITLE
refactor: update to new effect syntax

### PR DESCRIPTION
**Pull request by ninaandrup**
Related to https://github.com/flix/flix/issues/4377

### CODE REVIEW

File: src/effects.md [hunk 1]
```diff
@@ -104,7 +104,7 @@ function `h` that accepts two function arguments `f`
 and `g` of which at most one is impure:
 
 ```flix
-def h(f: a -> b & ef1, g: b -> c & (not ef1 or ef2)): Unit
+def h(f: a -> b \ ef1, g: b -> c \ { (not ef1) or ef2 }): Unit
```
Comment (ninaandrup) [inline src/effects.md:107]: I need some help to make this correct:
```
def h(f: a -> b & ef1, g: b -> c & (not ef1 or ef2)): Unit
```

**Comment by ninaandrup** [comment]
When I compiled the `mdbook` it automatically fixed some indentation and for some reason it changed the italic syntax from `*italic*` to `_italic_`.

File: src/effects.md (hunk 1 above)
Comment (magnus-madsen) [reply src/effects.md:107]: Ah, this example cannot be updated, so just leave it.

**Comment by magnus-madsen** [comment]
> When I compiled the `mdbook` it automatically fixed some indentation and for some reason it changed the italic syntax from `*italic*` to `_italic_`.

Okay, and this is with a new version of mdbook?

**Comment by ninaandrup** [comment]
> > When I compiled the `mdbook` it automatically fixed some indentation and for some reason it changed the italic syntax from `*italic*` to `_italic_`.
> 
> Okay, and this is with a new version of mdbook?

It could be because I used the newest version. I'm not sure. But it still looks the same, when I compile. But I can change it back, if the `*` is better for italic than `_` :-)

But from this [markdownguide](https://www.markdownguide.org/basic-syntax/#italic) it looks like both are good unless it is the middle of a word that needs to be italic.

**Comment by ninaandrup** [comment]
Are the keywords `Pure` and `Impure` still a part of the [`Reserved Identifiers`](https://doc.flix.dev/identifiers.html) ?

**Pull request merged**